### PR TITLE
feat(RELEASE-1054) fix linting issues in add fbc contribution task

### PR DESCRIPTION
**Pull request by happybhati**
This commit fixes the linting issue generated as a checkton warning
 in the `add-fbc-contribution` task and test files.

### CODE REVIEW

File: tasks/add-fbc-contribution/tests/test-add-fbc-contribution-hotfix.yaml [hunk 1]
```diff
@@ -19,11 +19,11 @@ spec:
           - name: setup-values
             image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
             script: |
-              #!/usr/bin/env sh
+              #!/usr/bin/env bash
```
Comment (jinqi7) [inline tasks/add-fbc-contribution/tests/test-add-fbc-contribution-hotfix.yaml:22]: Shall we change to use "bash" directly?

File: .github/workflows/lint.yaml
Comment (mmalina) [inline .github/workflows/lint.yaml:1]: Typo in commit message / PR title: warning

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -53,19 +53,12 @@ jobs:
...
-            echo "Checkton found issues"
-            exit 1
-          else
-            echo "Checkton found no issues"
-          fi
+          find-copies-harder: false
+          # Set to false to disable the differential check
+          differential: false 
+          # Add a regex to include only the files you want to check
+          include-regex: "tasks/add-fbc-contribution/.*\\.yaml"
```
Comment (mmalina) [inline .github/workflows/lint.yaml]: Opening a thread so that this isn't left here.

File: tasks/add-fbc-contribution/README.md [hunk 3]
```diff
@@ -14,6 +14,9 @@ Task to create a internalrequest to add fbc contributions to index images
 | targetIndex    | targetIndex value updated by update-ocp-tag task                                          | No       | -                    |
 | resultsDirPath | Path to results directory in the data workspace                                           | No       | -                    |
 
+## Changes in 3.2.1
+* Improved POSIX compliance and variable handling in the check-data-keys task
```
Comment (mmalina) [inline tasks/add-fbc-contribution/README.md]: It should first of all say that we're fixing checkton/shellcheck linting issues.

File: tasks/add-fbc-contribution/add-fbc-contribution.yaml [hunk 4]
```diff
@@ -69,106 +69,109 @@ spec:
             exit 1
         fi
 
-        echo -n "$(workspaces.data.path)/ir-$(context.taskRun.uid)-result.json" > $(results.requestResultsFile.path)
+        printf "%s" "$(workspaces.data.path)/ir-$(context.taskRun.uid)-result.json" > \
```
Comment (mmalina) [inline tasks/add-fbc-contribution/add-fbc-contribution.yaml]: Why move to `printf`? If it was about POSIX sh not supporting echo options, it should be fixed by moving to `bash`.

File: tasks/add-fbc-contribution/README.md (hunk 3 above)
Comment (mmalina) [reply tasks/add-fbc-contribution/README.md]: Also, why would we want to improve POSIX compliance? We're totally fine using bash and non-posix things.

File: tasks/add-fbc-contribution/tests/test-add-fbc-contribution-hotfix.yaml (hunk 1 above)
Comment (mmalina) [reply tasks/add-fbc-contribution/tests/test-add-fbc-contribution-hotfix.yaml:22]: @jinqi7 why? Usually `/usr/bin/env` is preferred. But it seems the main reason is portability and in our case we know that we're usually running in a rhel-flavour of linux, so it's true we could point to bash directly. But on the other hand why not use `#!/usr/bin/env` and have one less thing to worry about?

Comment (jinqi7) [reply tasks/add-fbc-contribution/tests/test-add-fbc-contribution-hotfix.yaml:22]: I am asking if we can change "sh" to "bash".   If so, it becomes easy. I don't know why  "sh" is used before.

Comment (mmalina) [reply tasks/add-fbc-contribution/tests/test-add-fbc-contribution-hotfix.yaml:22]: Then I misunderstood your question. We can definitely use `bash` and we do in many places. `sh` is fine too, if it suites your needs. But with shellcheck we are learning that many of the things we regularly use are not supported in strictly posix-compliant sh. Like `<<<` or `[[ ]]`.

Comment (jinqi7) [reply tasks/add-fbc-contribution/tests/test-add-fbc-contribution-hotfix.yaml:22]: Thanks for clarifying it.

Comment (happybhati) [reply tasks/add-fbc-contribution/tests/test-add-fbc-contribution-hotfix.yaml:22]: I agree with  Martin; using `bash`  is the best way to avoid warnings for regularly used expressions Like <<< or [[ ]].

File: tasks/add-fbc-contribution/README.md (hunk 3 above)
Comment (happybhati) [reply tasks/add-fbc-contribution/README.md]: Yes, I agree. Fixed.
Yes, true. In many cases, it's better to use `bash` to adhere to our code style.

File: .github/workflows/lint.yaml
Comment (happybhati) [reply .github/workflows/lint.yaml:1]: Fixed, good catch. I should change my keyboard.

File: tasks/add-fbc-contribution/add-fbc-contribution.yaml (hunk 4 above)
Comment (happybhati) [reply tasks/add-fbc-contribution/add-fbc-contribution.yaml]: I agree. I was fixing warnings in sequence and later discovered switching to `bash` is the right way. 
It's fixed now, thanks.

File: .github/workflows/lint.yaml
Comment (mmalina) [reply .github/workflows/lint.yaml:1]: It's better, but still not proper English:
`s/This commit fix the linting issue generated as checkton warning/This commit fixes the linting issues generated as checkton warnings/`

File: tasks/add-fbc-contribution/README.md [hunk 5]
```diff
@@ -14,6 +14,9 @@ Task to create a internalrequest to add fbc contributions to index images
 | targetIndex    | targetIndex value updated by update-ocp-tag task                                          | No       | -                    |
 | resultsDirPath | Path to results directory in the data workspace                                           | No       | -                    |
 
+## Changes in 3.2.1
+* fixing checkton/shellcheck linting issues in the task and test
```
Comment (mmalina) [inline tasks/add-fbc-contribution/README.md]: I wanted to say that you should capitalize Fixing. But then I noticed that we haven't really been consistent in this file. Would you mind capitalizing all of them? If not, it's ok. Just the first letter after each `* `

File: tasks/add-fbc-contribution/add-fbc-contribution.yaml (hunk 4 above)
Comment (mmalina) [reply tasks/add-fbc-contribution/add-fbc-contribution.yaml]: I still see some changes from echo to printf. To be clear, it's completely fine to use printf, but it seems unnecessary to change from echo to printf here. Usually echo is the first tool of choice.

File: tasks/add-fbc-contribution/README.md (hunk 5 above)
Comment (happybhati) [reply tasks/add-fbc-contribution/README.md]: Certainly, on it.

File: tasks/add-fbc-contribution/add-fbc-contribution.yaml (hunk 4 above)
Comment (happybhati) [reply tasks/add-fbc-contribution/add-fbc-contribution.yaml]: Ah, fixed. I completely agree; thanks!

File: tasks/add-fbc-contribution/README.md (hunk 5 above)
Comment (happybhati) [reply tasks/add-fbc-contribution/README.md]: Done!

Comment (mmalina) [reply tasks/add-fbc-contribution/README.md]: Thanks!

File: tasks/add-fbc-contribution/add-fbc-contribution.yaml
Comment (mmalina) [inline tasks/add-fbc-contribution/add-fbc-contribution.yaml:1]: It seems you still have two warnings here:
```
warning: Double quote to prevent globbing and word splitting.
    ┌─ tasks/add-fbc-contribution/add-fbc-contribution.yaml:131:17
    │
131 │         echo -n $timestamp > "$(results.buildTimestamp.path)"
    │                 ^^^^^^^^^^
    │
    = SC2086
    = Defect reference: https://github.com/koalaman/shellcheck/wiki/SC2086

warning: Double quote to prevent globbing and word splitting.
    ┌─ tasks/add-fbc-contribution/add-fbc-contribution.yaml:132:17
    │
132 │         echo -n $target_index > "$(results.requestTargetIndex.path)"
    │                 ^^^^^^^^^^^^^
    │
    = SC2086
    = Defect reference: https://github.com/koalaman/shellcheck/wiki/SC2086
```

Comment (happybhati) [reply tasks/add-fbc-contribution/add-fbc-contribution.yaml:1]: Yes, I noticed the same and fixed, thanks!

File: .github/workflows/lint.yaml
Comment (happybhati) [reply .github/workflows/lint.yaml:1]: Fixed thanks!

File: tasks/add-fbc-contribution/tests/test-add-fbc-contribution-pre-ga.yaml [hunk 6]
```diff
@@ -82,16 +82,18 @@ spec:
...
-              #!/usr/bin/env sh
+              #!/usr/bin/env bash
               #
               set -eux
 
               internalRequest="$(kubectl get internalrequest --sort-by=.metadata.creationTimestamp --no-headers | \
                 tac | tail -1)"
 
-              requestParams=$(kubectl get internalrequest ${internalRequest} -o jsonpath="{.spec.params}")
+              internalRequest=$(echo "${internalRequest}" | xargs)
```
Comment (johnbieren) [inline tasks/add-fbc-contribution/tests/test-add-fbc-contribution-pre-ga.yaml:92]: why this change?

Comment (happybhati) [reply tasks/add-fbc-contribution/tests/test-add-fbc-contribution-pre-ga.yaml:92]: I added quotes to `${internalRequest}` to satisfy Checkton :`"${internalRequest}"`
Once I added the quotes, the test started failing, and from logs, I found extra spaces after the `internal request `

```
[check-result : check-result] ++ tac
[check-result : check-result] ++ tail -1
[check-result : check-result] + internalRequest='iib-hwgz2         '
[check-result : check-result] ++ kubectl get internalrequest 'iib-hwgz2         ' -o 'jsonpath={.spec.params}'
[check-result : check-result] Error from server (NotFound): internalrequests.appstudio.redhat.com "iib-hwgz2         " not found
```

So, I used `xargs` to trim extra spaces.

Comment (johnbieren) [reply tasks/add-fbc-contribution/tests/test-add-fbc-contribution-pre-ga.yaml:92]: interesting, thanks for the explanation

**Review by johnbieren — DISMISSED** [review]

**Review by mmalina — DISMISSED** [review]

**Review by johnbieren — APPROVED** [review]
